src/dolphin/downloadDolphin.ts

```diff
@@ -145,105 +146,110 @@ async function installDolphin(
+      case "win32": {
+        // don't need to backup user files since our zips don't contain them
+        log(`Extracting to: ${dolphinPath}`);
+        const zip = new AdmZip(assetPath);
+        zip.extractAllTo(dolphinPath, true);
+        break;
       }
-
-      log(`Extracting to: ${dolphinPath}`);
-      try {
-        await extractDmg(assetPath, dolphinPath);
-      } catch (err) {
-        log("Could not extract dmg");
-        console.log(err);
+      case "darwin": {
+        await installDolphinOnMac(assetPath, dolphinPath, log);
+        break;
       }
-      await fs.readdir(dolphinPath, (err, files) => {
-        if (err) {
-          log(err.message);
-        }
-        if (files) {
-          files.forEach(async (file) => {
-            if (file !== "Slippi Dolphin.app") {
-              try {
-                await fs.unlink(path.join(dolphinPath, file));
-              } catch (err) {
-                fs.removeSync(path.join(dolphinPath, file));
-              }
-            }
-          });
-        }
-      });
-      const binaryLocation = path.join(dolphinPath, "Slippi Dolphin.app", "Contents", "MacOS", "Slippi Dolphin");
-      const userInfo = os.userInfo();
-      try {
-        await fs.chmod(path.join(dolphinPath, "Slippi Dolphin.app"), "777");
-        await fs.chown(path.join(dolphinPath, "Slippi Dolphin.app"), userInfo.uid, userInfo.gid);
-        await fs.chmod(binaryLocation, "777");
-        await fs.chown(binaryLocation, userInfo.uid, userInfo.gid);
-      } catch (err) {
-        log(`could not chmod/chown ${type} Dolphin\n${err}`);
+      case "linux": {
+        await installDolphinOnLinux(type, assetPath, dolphinPath, log);
+        break;
       }
-
-      // Move backed up User folder and user.json
-      if (alreadyInstalled) {
-        const oldUserFolder = path.join(backupLocation, "Slippi Dolphin.app", "Contents", "Resources", "User");
-        const newUserFolder = path.join(dolphinResourcesPath, "User");
-
-        if (await fs.pathExists(oldUserFolder)) {
-          log("moving User folder...");
-          await fs.move(oldUserFolder, newUserFolder);
-        } else {
-          log("no old User folder to move");
-        }
-        await fs.remove(backupLocation);
+      default: {
+        throw new Error(`Installing Netplay is not supported on this platform: ${process.platform}`);
       }
-      break;
     }
-    case "linux": {
-      // Delete the existing app image if there is one
-      try {
-        const dolphinAppImagePath = await findDolphinExecutable(type);
-        // Delete the existing app image if it already exists
-        if (await fileExists(dolphinAppImagePath)) {
-          log(`${dolphinAppImagePath} already exists. Deleting...`);
-          await fs.remove(dolphinAppImagePath);
-        }
-      } catch (err) {
-        log("No existing AppImage found");
-      }
+  } catch (err) {
+    log("Failed to install Dolphin...");
+    console.log(err);
+  }
```

Comment:
I don't understand why we want to catch this error here and then do nothing about it. Shouldn't the parent caller of the installDolphin function actually know if the install failed or not?

Reply:
yeah it should probably be handled in the caller and then show a toast on failure.